Python Runtime - The Final Countdown

bronsa opened the pull request.

romeovs committed:
Use normal title case

romeovs committed:
Add syntax highlighting to Python script on Transform page

romeovs committed:
Handle json-line output

romeovs committed:
Remove unused props

romeovs committed:
Rename PythonEditor to PythonQueryEditor

romeovs committed:
Move script rewriter to utils

romeovs committed:
Flatten Python transform page

romeovs committed:
Rename shared to common

romeovs committed:
Remove dead code

romeovs committed:
Reduce specialization in Transform editor

romeovs committed:
Remove unused props

romeovs committed:
Move PythonQueryEditor to top-level components

romeovs committed:
Nest Python editor components in PythonTransformEditor

romeovs committed:
Restructure Python transform editor

romeovs committed:
Split up component more

romeovs committed:
Add helpers for python editor

romeovs committed:
Disable default schema

romeovs committed:
Add test ids

romeovs committed:
Fix import path

romeovs committed:
Reuse transform save modal

romeovs committed:
Add e2e test for Python transform cretion

romeovs committed:
Fix types

romeovs committed:
Fix parameter

romeovs committed:
Add e2e tests for python common lib

romeovs committed:
Move python transform api to general transforms folder

romeovs committed:
Move type to metabase-api/types

romeovs committed:
Disable debugger for python transforms for now, allow to open it with query param

romeovs committed:
Disable debugger hotkey for python transforms

bronsa committed:
update changelog

bronsa committed:
add type->database-type unit tests

bronsa committed:
kondo

bronsa committed:
create/insert table

bronsa committed:
insert-from-jsonl test

bronsa committed:
Changes from cherry-pick branch

bronsa committed:
missing rename-tables for snowflake

bronsa committed:
extend rename-tables-test to more drivers

bronsa committed:
line breaks

bronsa committed:
add driver python transform feature

romeovs committed:
Remove cancelation code for now

bronsa committed:
handle review comments

romeovs committed:
Add more generic RunInfo

romeovs committed:
Remove unused LogOutput

bronsa committed:
kondo

romeovs committed:
Set up python runner in CI (#63566)

* Check out with submodules in CI

* Include python-runner in docker compose for tests

* Set up python runner in CI

* Fix import

* Update python-runner

bronsa committed:
feat: replace string->val with a more generic insert-col->val

romeovs committed:
Fix font sizes

romeovs committed:
Fix table layout

romeovs committed:
Revert "Remove unused LogOutput"

This reverts commit 90c07043449ca20fdfc126aec6e0feb5f8db4c64.

bronsa committed:
remove non transactional impls

bronsa committed:
chore: docsstrings

bronsa committed:
TODO

romeovs committed:
Add LogOutput back to runsection

romeovs committed:
Remove duplicate schema field

romeovs committed:
Use css instead of styled components

romeovs committed:
Fix LogOutput styling

bronsa committed:
fix sqls

bronsa committed:
type restriction.

crisptrutski committed:
Fix for empty input tables with LocalStack (#63639)

bronsa committed:
fix: order

bronsa committed:
s/keep/filter

crisptrutski commented on enterprise/backend/src/metabase_enterprise/transforms_python/python_runner.clj:
```diff
@@ -87,23 +87,18 @@
 
-(defn- root-type
+(defn root-type
+  "Supported type for rountrip/insertion"
```
typo in roundtrip :-D

bronsa committed:
typo

bronsa committed:
cherry pick driver code (#63628)

* cherry pick driver code

* update changelog

* add type->database-type unit tests

* kondo

* create/insert table

* insert-from-jsonl test

* missing rename-tables for snowflake

* extend rename-tables-test to more drivers

* line breaks

* add driver python transform feature

* handle review comments

* kondo

* feat: replace string->val with a more generic insert-col->val

* remove non transactional impls

* chore: docsstrings

* TODO

* extra test

* fix extra test

* test all base-types

* fix sqls

bronsa committed:
cherry pick driver code (#63628)

* cherry pick driver code

* update changelog

* add type->database-type unit tests

* kondo

* create/insert table

* insert-from-jsonl test

* missing rename-tables for snowflake

* extend rename-tables-test to more drivers

* line breaks

* add driver python transform feature

* handle review comments

* kondo

* feat: replace string->val with a more generic insert-col->val

* remove non transactional impls

* chore: docsstrings

* TODO

* extra test

* fix extra test

* test all base-types

* fix sqls

qnkhuat committed:
Proper CI setup: clone instead of sub modules (#63654)

romeovs committed:
Add transforms/python feature

romeovs committed:
Do not allow picking unsupported databases

romeovs committed:
Use slugify from metabase/lib/formatting

romeovs committed:
Add type for transform alias map

romeovs committed:
Fix some type imports

romeovs committed:
Refactor PythonDataPicker

romeovs committed:
Further refactors

crisptrutski committed:
Track versioning (#63534)

crisptrutski committed:
Rename setting for consistency

crisptrutski committed:
Remove staging creds from code

romeovs committed:
Use data selector for database picker

crisptrutski committed:
Missed a line somehow

bronsa committed:
instrument existing transform paths

romeovs committed:
Use data picker to pick tables

qnkhuat committed:
setup ci properly (#63693)

qnkhuat committed:
setup ci properly (#63693)

bronsa committed:
chore

romeovs committed:
Improve tests for python transform

bronsa committed:
maintain ordering

crisptrutski committed:
Merge branch 'master' into proj-python/s01e04

crisptrutski committed:
Merge branch 'merge-from-master-sep-18' into proj-python/s01e04

crisptrutski committed:
Yoink s3 code out

crisptrutski committed:
Super embarrassing

crisptrutski committed:
Fix lints and minor bug

crisptrutski committed:
Remove staging settings from defaults

crisptrutski committed:
Yoinked s3 code

bronsa committed:
Merge branch 'proj-python-master' into proj-python/s01e04

crisptrutski committed:
Test it

crisptrutski committed:
Better docstring for the most complex method

crisptrutski committed:
Better names

crisptrutski committed:
Fix-up rename

crisptrutski committed:
Fix whitespace

qnkhuat committed:
check feature flag for proper transform when create and run (#63641)

romeovs committed:
Use DatabaseId in python transform

crisptrutski committed:
Merge branch 's3-helpers' into proj-python/s01e04

crisptrutski committed:
Merge branch 'merge-s3-changes' into proj-python/s01e04

qnkhuat committed:
fix drivers ci

qnkhuat committed:
spin python runner for postgres app db tests as well

romeovs committed:
Remove source/draft ambiguity in types

romeovs committed:
Move library types to metabase-types/api

romeovs committed:
Remove api transforms

romeovs committed:
Remove vibe-comment

bronsa committed:
cherry pick changes from feature branch

romeovs committed:
Remove unused component

romeovs committed:
Validate source more

romeovs committed:
Move TableInput to separate file

romeovs committed:
Remove TableOption type

romeovs committed:
Simplify props

romeovs committed:
Move AliasInput to separate file

romeovs committed:
Remove unused props

romeovs committed:
Add border to empty results

bronsa committed:
Merge remote-tracking branch 'origin/master' into proj-python/s01e04

romeovs committed:
Fix import

romeovs committed:
Remove extension

romeovs committed:
Move draft back to new python transform page

romeovs committed:
Fix new python transform page nesting

crisptrutski committed:
S3 transport layer for python-runner (#63697)

crisptrutski committed:
Merge branch 'proj-python/feature-branch' into proj-python/s01e04

bronsa committed:
chore: update drivers matrix to type restriction

crisptrutski committed:
Remove rich comments

bronsa committed:
wider buckets

bronsa committed:
remove nil handling

bronsa committed:
warn on nil bytes/rows

crisptrutski committed:
Add dependencies

bronsa committed:
add stage-type mapping

crisptrutski committed:
Make s3 a driver test to get localstack

wotbrew committed:
Fix SocketException handling on interrupt during logging (#63705)

bronsa committed:
pull out type restriction helper

bronsa committed:
fix driver matrix

bronsa committed:
tests are passing

bronsa committed:
linter fixes

bronsa committed:
typo

bronsa committed:
Merge remote-tracking branch 'origin/add-s3-deps' into proj-python/s01e04

bronsa committed:
comment out postgres driver tests during appddb workflow

bronsa committed:
fix test

crisptrutski committed:
Simplify control flow a bit

bronsa committed:
missed

bronsa committed:
fix case

crisptrutski committed:
Disable s3 tests when run in the wrong group

crisptrutski committed:
Merge branch 'add-s3-deps' into proj-python/s01e04

bronsa committed:
h2 ain't enabled for now

crisptrutski committed:
Remove stray stub

crisptrutski committed:
Merge branch 'merge-s3-fixes' into proj-python/s01e04

bronsa committed:
log/warnf is a macro, we can't with-redefs. how did this test ever work?

bronsa committed:
disable h2 here too

bronsa committed:
kondo

bronsa committed:
fix

bronsa committed:
fix: missing feature

bronsa committed:
fix: proper sleep

bronsa committed:
fix broken test

bronsa committed:
fix yet another test that never worked

bronsa committed:
remove exotic maria type not in CI

bronsa committed:
wait for fields

bronsa committed:
switch to wait-for-table

bronsa committed:
less fancy for maria

bronsa committed:
shorter name

crisptrutski committed:
Add s3 dependencies (#63706)

bronsa committed:
shorter

bronsa committed:
Merge remote-tracking branch 'origin/proj-python/feature-branch' into proj-python/s01e04

bronsa committed:
fix: order

bronsa committed:
eh, that was worse, we're probably nondeterministic somewhere, TODO

bronsa committed:
port prometheus/instrumentation (#63631)

* port prometheus metrics

* port instrumentation

* instrument existing transform paths

* wider buckets

* remove nil handling

* warn on nil bytes/rows

* add stage-type mapping

* typo

bronsa committed:
Merge remote-tracking branch 'origin/proj-python/feature-branch' into proj-python/s01e04

romeovs committed:
Simpify new transform page structure

romeovs committed:
Implement data picker according to designs

romeovs committed:
Further adjust to design file

romeovs committed:
Fix some e2e tests

wotbrew committed:
Reorder the final message-log flush to occur before the succeed call

Should fix issues in tests that block on until a 'terminal' state, log output could be late and cause the test to fail.

romeovs committed:
Clone python runner when running CI

romeovs committed:
Avoid starting the container in e2e tests twice

qnkhuat committed:
Transforms python tests with its own CI (#63744)

wotbrew committed:
Add python runner

bronsa committed:
fixes for clickhouse

qnkhuat committed:
split CI into separate jobs (#63746)

wotbrew committed:
Kondo

bronsa committed:
fix test

wotbrew committed:
Traceback fix

wotbrew committed:
Remove dead code

qnkhuat committed:
start the runner by default

bronsa committed:
fix

bronsa committed:
bump

wotbrew committed:
Cancellation not used yet

wotbrew committed:
Remove unused keys from fields-metadata

bronsa committed:
fix for redshift

bronsa committed:
Merge remote-tracking branch 'origin/proj-python/feature-branch' into proj-python/s01e04

bronsa committed:
cherry pick driver fixes

bronsa committed:
order

qnkhuat committed:
start for postgres and mysql as well

bronsa committed:
clickhouse fixes

bronsa committed:
clickhouse fixes

bronsa committed:
relax for snowflake

bronsa committed:
relax for snowflake

bronsa committed:
fix

bronsa committed:
typo

bronsa committed:
typo

bronsa committed:
exclude transform-python-test from appdb

bronsa committed:
further relax

qnkhuat committed:
exclude transforms-python-test as well

bronsa committed:
try exclude-tag here too

bronsa committed:
disable flake

wotbrew committed:
Add python_runner.clj (#63747)

* Add python runner

* Kondo

* Traceback fix

* Remove dead code

* start the runner by default

* Cancellation not used yet

* Remove unused keys from fields-metadata

* cherry pick driver fixes

* start for postgres and mysql as well

* clickhouse fixes

* relax for snowflake

* typo

* exclude transform-python-test from appdb

* exclude transforms-python-test as well

* try exclude-tag here too

---------

Co-authored-by: Ngoc Khuat <qn.khuat@gmail.com>
Co-authored-by: Nicola Mometto <nicola.mometto@metabase.com>

bronsa committed:
Merge branch 'proj-python/runner' into proj-python/s01e04

bronsa committed:
Merge remote-tracking branch 'origin/proj-python/feature-branch' into proj-python/s01e04

qnkhuat committed:
execute.clj

qnkhuat committed:
with drivers test and execute test

qnkhuat committed:
with wait tables

qnkhuat committed:
missing utils

qnkhuat committed:
/s/open/start

qnkhuat committed:
fix failing tests due to getting the wrong source and target

bronsa committed:
kondo

bronsa committed:
move cancel-python-code-http-call to python-runner

qnkhuat committed:
Dan's feedback

qnkhuat committed:
we have 2 implementstation of cancel

bronsa committed:
Merge branch 'python-transform-execute' into proj-python/s01e04

qnkhuat committed:
execute.clj (#63755)

* execute.clj

* with drivers test and execute test

* with wait tables

* missing utils

* /s/open/start

* fix failing tests due to getting the wrong source and target

* kondo

* move cancel-python-code-http-call to python-runner

* Dan's feedback

* we have 2 implementstation of cancel

* and 2 db-routing-enabled?

---------

Co-authored-by: Nicola Mometto <nicola.mometto@metabase.com>

bronsa committed:
Merge remote-tracking branch 'origin/proj-python/feature-branch' into proj-python/s01e04

romeovs committed:
Hide python runner admin route when user does not have the feature for it

romeovs committed:
Remove unused variables

romeovs committed:
Move Python runner amdin page to transforms plugin

romeovs committed:
Do not show Python library link when the user does not have python transforms support

romeovs committed:
Hide Python transforms features when the user does not have transforms-python token

romeovs committed:
Use useHasTokenFeature hook where possible

romeovs committed:
Add transforms-python feature

romeovs committed:
Fix e2e tests

romeovs committed:
Do not include python-runner in test docker-compose yet

qnkhuat committed:
Transforms python APIs (#63762)

* transforms python api

* new tests pass locally

* clean up the db id utils

* move to util

* fix: missing refer

* chore: missing tests/kondo

* kondo

* more kondo

* fix: make function resilient

* eh, validate on old

* on new

* Add python transform support for jobs (#63764)

* Add python transforms job support

* Fix whitespace in when

* Avoid churn due to new local

* Add feature test

* Add ignore for logs (there might be away to avoid this)

* Use new target-database-id util

* Remove duplicated fn

---------

Co-authored-by: Nicola Mometto <nicola.mometto@metabase.com>
Co-authored-by: Dan Stone <wotbrew@gmail.com>

wotbrew committed:
Merge branch 'proj-python/feature-branch' into proj-python/s01e04

# Conflicts:
#	enterprise/backend/src/metabase_enterprise/transforms/api.clj
#	enterprise/backend/src/metabase_enterprise/transforms/canceling.clj
#	enterprise/backend/src/metabase_enterprise/transforms/jobs.clj
#	enterprise/backend/src/metabase_enterprise/transforms/ordering.clj
#	enterprise/backend/src/metabase_enterprise/transforms/util.clj
#	enterprise/backend/test/metabase_enterprise/transforms/api_test.clj
#	enterprise/backend/test/metabase_enterprise/transforms/test_util.clj

wotbrew committed:
Remove dead code

wotbrew committed:
Fix missing require

wotbrew committed:
Fix private

romeovs committed:
Tag tests for python transforms with @transforms-python

romeovs committed:
Add helper to update Python library

romeovs committed:
Add check for python output

romeovs committed:
Fix Revert on python library when it does not exist

romeovs committed:
Fix e2e test helper

romeovs committed:
Add python runner settings in CI

romeovs committed:
Translate pandas definitions

romeovs committed:
Add AnsiLogs component

romeovs committed:
Use AnsiLogs in Logs components

romeovs committed:
Use AnsiLogs in LogOutput component

romeovs committed:
Remove token feature gate in enterprise folder

romeovs committed:
Return alias instead of throwing

romeovs committed:
Check type in isPythonTransform assertion

romeovs committed:
Use isResourceNotFoundError

romeovs committed:
Make transforms-python it's own plugin

bronsa committed:
support bigints
